pkg/deploy/strategy/rolling/rolling.go

```diff
@@ -150,17 +151,25 @@ func (s *RollingDeploymentStrategy) Deploy(from *kapi.ReplicationController, to
 	//
 	// Related upstream issue:
 	// https://github.com/kubernetes/kubernetes/pull/7183
-	to, err = s.client.ReplicationControllers(to.Namespace).Get(to.Name)
+	err = wait.Poll(1*time.Second, 10*time.Second, func() (done bool, err error) {
+		existing, err := s.client.ReplicationControllers(to.Namespace).Get(to.Name)
+		if err != nil {
+			return false, fmt.Errorf("couldn't look up deployment %s: %s", deployutil.LabelForDeployment(to), err)
```

Comment:
I refactored this so that notfound errors are considered fatal while all others are logged and retried. PTAL.